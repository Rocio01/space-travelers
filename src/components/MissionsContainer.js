import React from 'react'

export default function MissionsContainer() {
  return (
    <div>
      
    </div>
  )
}
