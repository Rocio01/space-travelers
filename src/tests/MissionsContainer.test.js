import React from 'react';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider, useDispatch } from 'react-redux';
import missionsReducer, { fetchMissions } from '../redux/missions/missions';
import '@testing-library/jest-dom/extend-expect';
import MissionsContainer from '../components/MissionsContainer';
import store from '../redux/configureStore';

jest.mock('./mocks/api.js');


// const dispatch = useDispatch();
// describe('Missions tests', () => {
//   it('Missions page matches snapshot', async () => {
//     await fetchMissions(dispatch);
//     const missionsPage = renderer
//       .create(<Provider store={store}><MissionsContainer /></Provider>)
//       .toJSON();
//     expect(missionsPage).toMatchSnapshot();
//   });
// });

describe('missions', () => {
  test('matches MissionsContainer component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MissionsContainer />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should return the initial state', () => {
    expect(missionsReducer(undefined, {})).toEqual({ missions: [] });
  });
});
