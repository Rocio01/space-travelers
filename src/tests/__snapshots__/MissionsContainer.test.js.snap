// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missions matches MissionsContainer component snapshot 1`] = `
<div
  className="container mt-5"
  id="missions-container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th
          className="col-1"
        >
          mission
        </th>
        <th>
          Description
        </th>
        <th
          className="col-2"
        >
          Status
        </th>
        <th
          className="col-2"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
